openEditFormとeditTodoの内容修正

kaayaa00648 opened the pull request.

kaayaa00648 committed:
initial commit

Junkichi89 commented:
@kaayaa00648 
お疲れ様です。

プルリクエストの使い方について、
あくまで参考ですが、自分が依頼している時はこのような形でまずは説明を書いています。
![image](https://user-images.githubusercontent.com/65665381/132220245-66d1b763-3edd-456e-8ec8-df7761e62e86.png)

今回は、個人用なので、実現したいことと現状理解できていないことやどの部分のコードを確認してほしいなど記載してもらえるといろんな回答者から意見をもらいやすいかなと思います。（自分は内容を把握しているので大丈夫です！w）

https://docs.google.com/spreadsheets/d/11cGw_2ZxroXRoDMjJVmyXEwG38Kz8Llojm-btS0CTIs/edit#gid=575507459&range=A29
普段は、こちらのシートの質問フォーマットの必要な部分だけ抜き出してまとめてもらえるといいのかなと思います。

あと、自分の考えとしては、プルリクエストというものはコードを確認してもらうためだけのものではなく、コードを通してコミュニケーションをする場だと考えているので、できる限り自分が実現したいことやうまくいかないところを記述して使ってもらえるとより効果的に使えるのではないかなと思います。

また、いただいたコメントに対して質問などができるのもプルリクエストのいいところですね！
では、今から、コードについてレビューしておきます。


可能であれば、自分の記録用に実現したいことなどまとめてもらえると嬉しいです。

Junkichi89 commented on src/App.jsx:
```diff
@@ -110,12 +113,12 @@ export const App = () => {
+                <li>{newTitle}</li>
                 <button onClick={() => onClickComplete(index)}>完了</button>
-                <button onClick={openEditForm}>編集</button>
+                <button onClick={() => openEditForm(index)}>編集</button>
```
ここが一つの問題でした。
すみません、多分自分が連絡した内容、ここができてれば、大丈夫でした。
なので①はクリアできてそうです！

Junkichi89 commented on src/App.jsx:
```diff
@@ -63,14 +67,13 @@ export const App = () => {
   const editTodo = () => {
-    // newEditTodos[editIndex].title = newTitle;
-    setIncompleteTodos((todo, editIndex) => {
+    setIncompleteTodos((editIndex) => {
```
```
setIncompleteTodos((editIndex) => {　
console.log(editIndex)
```

でeditIndexの値を確認してみてください。

Junkichi89 commented on src/App.jsx:
```diff
@@ -110,12 +113,12 @@ export const App = () => {
             return (
-              <div key={todo} className="list-row">
-                <li>{todo}</li>
+              <div key={newTitle} className="list-row">
```
この問題を解決するには、何通りか方法があるかと思いますが、
現状はこのままでもいいかと思いますが、
編集の実装が完成できた後にもう一度、振り返って、このエラーが出ないようにしてもらえるといいかなと思います。
一旦はスルーでOKです。（編集に集中してほしいので・・・、余裕あったら考えてみてください。）
![image](https://user-images.githubusercontent.com/65665381/132223902-efd41e16-ba6a-4f25-9238-8f69c84ae166.png)

Junkichi89 commented on src/App.jsx:
```diff
@@ -110,12 +113,12 @@ export const App = () => {
         <p className="title">未完了のTODO</p>
         <ul>
-          {incompleteTodos.map((todo, index) => {
+          {incompleteTodos.map((newTitle, index) => {
```
ちなみに、ここってnewTitleでもtodoでも同じって知ってますか？
MDNは難しいかもですが、mapの使い方を理解してもらった方がいいですね、あとはコールバック関数とかですかね。

https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/map

一応、説明というか少しでも参考になればなんですが、
`incompleteTodos.map((newTitle, index) => {`
ここでやっていることを簡単に説明すると
仮定として、
incompleteTodosの値が`["家事", "掃除", "料理"]`
だったとします、するとこのnewTitleに入ってくる値というのはまず初めに"家事"、次に"掃除"、　"料理"という値が順番に入ってきて、新しい配列が作成されます。
つまり、このnewTitleというものの正体としては、incompleteTodos[0] = "家事",
incompleteTodos[1] = "掃除"、
incompleteTodos[2] = "料理"
を簡単に表現しているものくらいに認識してもらえるといいのかなと思います。
（自分書くのめんどくさいからくらいにしか思ってませんが、何か他にちゃんとした理由があると思います・・・）
少し説明わかりづらくすみません。

なので、結論としては、　newTitleであっても、todoであっても同じ意味になってしまいます。

Junkichi89 reviewed (commented):
@kaayaa00648 
何点か行き違いをしてしまっていみたいで申し訳ないです。

自分から言えることとしては、もう少し、mapメソッドやuseStateで使われる変数の値の確認をしっかりと行った方がいいかと思います。

自分が勉強したての頃に教えてもらった方法としては、変数を使うときは全てconsole.logで確認するくらいにするといいよと教わりました。
そこで気をつけないといけないこととしては、全部の変数にconsole.logを入れるとどこのlogかわからなくなってしまうので、
`console.log("ここは〜の変数", 確認したい変数)`みたいにしてどこの変数なのか把握するのも大事ですね。

コードが複雑になってくると変数に入っている値が意図しない値になってしまうことが多々あるので、まずはどんな値が入っているか確認する癖をつけると今後スムーズにいくかもしれませんね。

グダグタとコメントしてしまいましたが、少しでも参考になればと思います。
おかしいところや説明がわかりづらいところなどありましたら、気軽にslackで連絡ください〜

kaayaa00648 committed:
編集ボタン処理と編集を保存ボタン処理を修正
目的
編集ボタンの追加・実装

望んでいるフィードバック
・コードに問題がないか

※今回codesandbox使用しております。

お手数をおかけしますが、よろしくお願いいたします。
